drivers/i3c/i3c_ccc.c

```diff
@@ -591,3 +591,100 @@ int i3c_ccc_do_getcaps(const struct i3c_device_desc *target,
 out:
 	return ret;
 }
+
+int i3c_ccc_do_getmxds(const struct i3c_device_desc *target,
+			 union i3c_ccc_getmxds *mxds,
+			 enum i3c_ccc_getmxds_fmt fmt,
+			 enum i3c_ccc_getmxds_defbyte defbyte)
+{
+	struct i3c_ccc_payload ccc_payload;
+	struct i3c_ccc_target_payload ccc_tgt_payload;
+	uint8_t defining_byte;
+	uint8_t data[5];
+	uint8_t len;
+	int ret;
+
+	__ASSERT_NO_MSG(target != NULL);
+	__ASSERT_NO_MSG(target->bus != NULL);
+	__ASSERT_NO_MSG(mxds != NULL);
+
+	ccc_tgt_payload.addr = target->dynamic_addr;
+	ccc_tgt_payload.rnw = 1;
+	ccc_tgt_payload.data = &data[0];
+
+	if ((fmt == GETMXDS_FORMAT_1) || (fmt == GETMXDS_FORMAT_2)) {
+		/* Could be 2 or 5 Data Bytes Returned */
+		ccc_tgt_payload.data_len = sizeof(((union i3c_ccc_getmxds *)0)->fmt2);
```

Comment:
not really a fan of using sizeof() here (and others), IMHO, makes code less readable than having eg a few defines.

Reply:
i guess I have to beg to differ here... but yeah.. it is an opinion thing even for me here as well

I prefer this rather than a macro which points to a magic number

Reply:
can you guarantee that eg structs won't have any holes?

Reply:
the struct is `__packed`